src/shared/utilities/timeoutUtils.ts

```diff
@@ -200,28 +238,30 @@ export function waitTimeout<T, R = void, B extends boolean = true>(
         promise = promise()
     }
 
-    return Promise.race([promise, timeout.timer])
-        .then(obj => {
-            if (obj !== undefined) {
-                return obj
-            }
-            if ((opt.allowUndefined ?? true) !== true) {
-                throw new Error(TIMEOUT_UNEXPECTED_RESOLVE)
-            }
-            return undefined as any
-        })
-        .catch(err => {
-            if (opt.onExpire && (err as Error).message === TIMEOUT_EXPIRED_MESSAGE) {
-                return opt.onExpire()
-            }
-            if (opt.onCancel && (err as Error).message === TIMEOUT_CANCELLED_MESSAGE) {
-                return opt.onCancel()
-            }
-            throw err
-        })
+    const result = await Promise.race([promise, timeout.promisify()])
+        .catch(e => (e instanceof Error ? e : new Error(`unknown error: ${e}`)))
```

Comment:
What cases would generate a non-error in the catch?

Reply:
None that I know of. Was getting tired of writing `(e as Error)...`. Better just to do the 'real' check if we're going to cast it.